### PR TITLE
Allow Promise to be passed to ArrayField.verbose_name

### PR DESCRIPTION
**Pull request by ljodal**
This expands the interface of ArrayField to allow lazy translations to be passed to the verbose_name field. Seems like this was missed in #1139

### CODE REVIEW

File: django-stubs/contrib/postgres/fields/array.pyi [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ArrayField(CheckFieldDefaultMixin, Field[_ST, _GT]):
         base_field: Field,
         size: Optional[int] = ...,
         *,
-        verbose_name: Optional[Union[str, bytes]] = ...,
+        verbose_name: Optional[Union[_StrOrPromise, bytes]] = ...,
         name: Optional[str] = ...,
```
Comment (ljodal) [inline django-stubs/contrib/postgres/fields/array.pyi]: Should name also accept _StrOrPromise?

Comment (sobolevn) [reply django-stubs/contrib/postgres/fields/array.pyi]: Yes, I think so

File: django-stubs/contrib/postgres/fields/array.pyi [hunk 2]
```diff
@@ -27,7 +27,7 @@ class ArrayField(CheckFieldDefaultMixin, Field[_ST, _GT]):
         base_field: Field,
         size: Optional[int] = ...,
         *,
-        verbose_name: Optional[Union[str, bytes]] = ...,
+        verbose_name: Optional[Union[_StrOrPromise, bytes]] = ...,
```
Comment (intgr) [inline django-stubs/contrib/postgres/fields/array.pyi]: Not really in the scope of this PR, but I can't help to ask, why is `bytes` even allowed here? Makes no sense IMO. Is this for Python 2 compatibility?

The commit message adding this simply states "fixes" d43cb1fcd7961ca620aaf1d67949baf2e3c040f3 (?!)

Comment (sobolevn) [reply django-stubs/contrib/postgres/fields/array.pyi]: I think it might be some legacy thing, I am totally fine with removing it.

Comment (intgr) [reply django-stubs/contrib/postgres/fields/array.pyi]: OK I can open a PR for that after this is merged, if ljodal doesn't get to it first.

File: django-stubs/contrib/postgres/fields/array.pyi (hunk 1 above)
Comment (ljodal) [reply django-stubs/contrib/postgres/fields/array.pyi]: Aight, updated that as well

File: django-stubs/contrib/postgres/fields/array.pyi (hunk 2 above)
Comment (ljodal) [reply django-stubs/contrib/postgres/fields/array.pyi]: I removed the bytes while fixing `name` :)

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks!

**Comment by intgr** [comment]
Hrm. The `name` argument of all other fields is still typed as `Optional[str]` though.

And I think it doesn't makes sense for `Field.name` to ever be `Promise`. It determines the attribute of the `Model` instance and the database field name, it'll be embedded in migrations and database tables, etc.

**Comment by ljodal** [comment]
Oh, oops, guess we should revert that bit then 😬

**Pull request merged**